iati/tests/test_validator.py

```diff
@@ -846,16 +833,67 @@ def test_validation_codelist_vocab_user_defined_with_uri_unreadable(self, schema
         assert iati.validator.is_valid(data, schema_sectors)
 
 
+class TestValidateRulesets(object):
+    """A container for tests relating to validation of Rulesets."""
+
+    def test_basic_validation_ruleset_valid(self, schema_ruleset):
+        """Perform data validation against valid IATI XML that is valid against the Standard Ruleset."""
+        data = iati.core.tests.utilities.load_as_dataset('valid_std_ruleset')
+
+        assert iati.validator.is_xml(data.xml_str)
+        assert iati.validator.is_iati_xml(data, schema_ruleset)
+        assert iati.validator.is_valid(data, schema_ruleset)
+
+    @pytest.mark.parametrize("invalid_xml_file", ['invalid_std_ruleset_bad_date_order', 'invalid_std_ruleset_bad_identifier', 'invalid_std_ruleset_does_not_sum_100', 'invalid_std_ruleset_missing_sector_element'])
+    def test_basic_validation_ruleset_invalid(self, schema_ruleset, invalid_xml_file):
+        """Perform data validation against valid IATI XML that does not conform to the Standard Ruleset."""
+        data = iati.core.tests.utilities.load_as_dataset(invalid_xml_file)
+
+        assert iati.validator.is_xml(data.xml_str)
+        assert iati.validator.is_iati_xml(data, schema_ruleset)
+        assert not iati.validator.is_valid(data, schema_ruleset)
```

Comment:
As well as checking this, for each file it should be ensured that the correct `error_or_warning_by_name` is raised.